File: app/src/main/java/com/arcuscomputing/dictionary/menu/IArcusMenu.java
package com.arcuscomputing.dictionary.menu;


import android.view.Menu;
import android.view.MenuItem;

import com.arcuscomputing.dictionary.ArcusSearchActivity;

public interface IArcusMenu {

    // MENU
    int MENU_SEARCH = 0;

    int MENU_SETTINGS = 1;
    int MENU_HELP = 2;
    int MENU_FAVOURITES = 3;
    int MENU_RANDOM = 4;

    int MENU_ALPHA_SORT = 5;
    int MENU_DATE_SORT = 6;
    int MENU_CLEAR_FAVOURITES = 7;
    int MENU_EMAIL_FAVOURITES = 8;

    //MENU ORDER
    int MENU_SEARCH_INDEX = 0;
    int MENU_FAVOURITES_INDEX = 1;
    int MENU_SETTINGS_INDEX = 2;
    int MENU_HELP_INDEX = 3;

    int MENU_RANDOM_INDEX = 4;

    int MENU_ALPHA_SORT_INDEX = 5;
    int MENU_DATE_SORT_INDEX = 6;
    int MENU_CLEAR_FAVOURITES_INDEX = 7;
    int MENU_EMAIL_FAVOURITES_INDEX = 8;

    // MENU GROUP
    int MENUGROUP_INFO = 0;
    int MENUGROUP_ACTIONS = 1;


    // CONTEXT
    int CONTEXT_GOOGLE_DICTIONARY = 0;
    int CONTEXT_WIKITIONARY = 1;

    boolean onCreateOptionsMenu();

    void setMainMenuItemsVisible(boolean visible);

    void setFavouritesMenuItemVisible(boolean visible);

    boolean onOptionsItemSelected(MenuItem item, ArcusSearchActivity activity);

    Menu getMenu();

    void setMenu(Menu menu);
}

File: app/src/main/java/com/arcuscomputing/dictionary/menu/IArcusMenu.kt
package com.arcuscomputing.dictionary.menu

import android.view.Menu
import android.view.MenuItem
import com.arcuscomputing.dictionary.ArcusSearchActivity

interface IArcusMenu {

    var menu: Menu

    fun onCreateOptionsMenu(): Boolean
    fun setMainMenuItemsVisible(visible: Boolean)
    fun setFavouritesMenuItemVisible(visible: Boolean)
    fun onOptionsItemSelected(item: MenuItem, activity: ArcusSearchActivity): Boolean

    companion object {
        const val MENU_SEARCH = 0
        const val MENU_SETTINGS = 1

        const val MENU_FAVOURITES = 2
        const val MENU_ALPHA_SORT = 3
        const val MENU_DATE_SORT = 4
        const val MENU_CLEAR_FAVOURITES = 5
        const val MENU_EMAIL_FAVOURITES = 6

        const val MENU_SEARCH_INDEX = 0
        const val MENU_FAVOURITES_INDEX = 1
        const val MENU_SETTINGS_INDEX = 2
        const val MENU_ALPHA_SORT_INDEX = 3
        const val MENU_DATE_SORT_INDEX = 4
        const val MENU_CLEAR_FAVOURITES_INDEX = 5
        const val MENU_EMAIL_FAVOURITES_INDEX = 6

        const val MENUGROUP_INFO = 0
        const val MENUGROUP_ACTIONS = 1
    }
}
